fixup

sevu opened the pull request.

sevu committed:
Improve scenario code

take over things from mainline
load some code not when entering game creation mode, but when starting a game
handle terrain codes better
different lock settings than mainline
activate force_lock_settings

sevu committed:
fixup 94f3ca1

sevu committed:
statisfy wmllint

rwallac commented on scenarios/4p_A_New_Land.cfg:
```diff
@@ -437,24 +504,30 @@
+            canrecruit=no # workaround
             [filter_location]
-                terrain=C*,K*
+                terrain=C*^*,K*^* # FIXME: This causes AI units to destroy their own castles too.
```
Do you mean that the ai units in the keep will sabotage their own keep if there are player units blocking them from moving further? I feel this would make it too easy. Or is this already in the game, and it's not really as big a of a deal as I think?

sevu replied:
They sabotage their own castle if they move onto one hex for fighting. (recruiting is not affected).
The bug was introduced 10 months ago, and was halfway resolved by itself when the movement restrictions from leaders were added from mainline.

I pushed a new commit, which doesn't destroy castles in a 3 hex radius from the keep, though this includes by the player build ones. Only filters with hex coordinates is can be be more precise.

sevu committed:
better terrain destruction code

Adresses the FIXMEs
It's more universal than before, it would as well work if new ways to build
things are added, e.g. farms on other terrian or villges inside castles.
The AI won't destroy casles around it's keep, that includes the ones build by
players there.

sevu committed:
remove unused macros

closes #21

sevu committed:
terrain code adjustments

merfolk are the only ones who can remove lilies
also added an explanaiton to terrain codes
finished rewriting of buildings to the macro and
use more variations for buildings

@Lukc great idea

closes #40

sevu committed:
Destruction takes different types into account, and changes
the terrain accordingly.
Buildings which were not in grassland have an advantage, as the terrain
does not need to be cleared afterwards.

The remove Paths action won't conflict with clearing the terrain anymore.
